.github/workflows/ci-repo.yml

```
@@ -39,43 +51,135 @@ jobs:
         run: |
-          cd vscode
-          npm version
           npm ci
 
       - name: Lint sources
-        run: |
-          cd vscode 
-          npm run lint
+        working-directory: ./vscode
+        run: npm run lint
+
+  test:
+    name: Test (${{ matrix.arch }})
+    runs-on: ${{ matrix.os }}
+    needs: build
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            arch: linux
+          - os: macos-latest
+            arch: macos
+          - os: windows-latest
+            arch: windows
 
-      - name: Build
-        run: |
-          cd vscode
-          npm run package
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-  vscode-tests:
-    runs-on: ubuntu-latest
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: "vscode/.nvmrc"
+
+      - name: Install dependencies
+        working-directory: ./vscode
+        run: npm ci
+
+      # Run tests on Linux
+      - name: Run tests (Linux)
+        if: matrix.arch == 'linux'
+        working-directory: ./vscode
+        run: 
+          xvfb-run -a npm run test
+
+      # Run tests on macOS
+      - name: Run tests (macOS)
+        if: matrix.arch == 'macos'
+        working-directory: ./vscode
+        run: npm test
+
+      # Run tests on Windows
+      - name: Run tests (Windows)
+        if: matrix.arch == 'windows'
+        working-directory: ./vscode
+        shell: cmd
+        run: npm test
+
+  package:
+    name: Package (${{ matrix.arch }})
+    runs-on: ${{ matrix.os }}
+    needs: test
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            arch: linux
+          - os: macos-latest
+            arch: macos
+          - os: windows-latest
+            arch: windows
+      max-parallel: 3
 
     steps:
-      - name: Check-out repository
+      - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Use Node.js from .nvmrc
+      - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
           node-version-file: "vscode/.nvmrc"
 
-      - name: Cache extension node_modules
-        uses: actions/cache@v3
+      - name: Install dependencies
+        working-directory: ./vscode
+        run: npm ci
+
+      - name: Build Package
+        working-directory: ./vscode
+        run: npm run package
+
+      - name: Generate .vsix package
+        working-directory: ./vscode
+        run: |
+          npm install @vscode/vsce
+          npx vsce package
+
+      - name: Upload VSIX artifact
+        uses: actions/upload-artifact@v4
         with:
-          path: vscode/node_modules
-          key: ${{ runner.os }}-node-${{ hashFiles('vscode/package-lock.json') }}
+          name: vscode-extension-${{ matrix.arch }}
+          path: "vscode/*.vsix"
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    needs: package
+    if: github.ref == 'refs/heads/main'
```

Update this to run on shouldRelease
```suggestion
    if: |
      github.ref == 'refs/heads/main' &&
      (github.event.inputs.shouldRelease == 'true' || github.event.inputs.shouldRelease == true)
```

---

updated